ballista/executor/src/flight_service.rs

```diff
@@ -103,18 +105,22 @@ impl FlightService for BallistaFlightService {
             executor_id = self.executor_id,
             job_id, stage_id, partition, path, "fetching shuffle partition"
         );
-        let file = File::open(path.as_str()).map_err(|e| {
+        let file = File::open(path.as_str()).await.map_err(|e| {
             Status::internal(format!("Failed to open partition file at {path}: {e:?}"))
         })?;
 
-        let reader = StreamReader::try_new(file, None).map_err(from_arrow_err)?;
+        let mut reader = AsyncStreamReader::try_new(file.compat(), None)
+            .await
+            .map_err(from_arrow_err)?;
         let schema = reader.schema();
 
         let (tx, rx) = channel(2);
 
         let executor_id = self.executor_id.clone();
-        task::spawn_blocking(move || {
-            if let Err(e) = read_partition(job_id, stage_id, partition, reader, tx) {
+        task::spawn(async move {
```

Comment:
We wouldn't need to spawn this anymore since it's async right?

Reply:
I think we want to spawn a sender on a separate task as we don't need to wait until we consume all underlying stream

Reply:
It's always going to consume the entire underlying stream. That's what this endpoint is doing. The only reason it was spawned before was because the file reader was blocking, but with the tokio fs operations and the async reader it's unneccessary